src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+          ),
+          toTokenAddress: hasEnoughPLR ? plrDaoMemberNFT.address : plrDaoAsset.address,
+          fromTokenAddress: selectedFromAsset.address,
+        });
+        return offers;
+      } catch (e) {
+        //
+      }
+    }, 200),
+    [sdk, selectedFromAsset, amount, selectedFromNetwork, accountAddress, selectedAccountType]
+  );
+
+  useEffect(() => {
+    accountAddress || providerAddress && sdk?.getNftList({
+      account: accountAddress || providerAddress,
+    }).then((output:NftList) => {
+      if(output?.items?.length) {
+        setIsNFTMember(true);
+      }
+    });
+    
+  }, [sdk, accountAddress, providerAddress]);
+
+  useEffect(() => {
+    // this will ensure that the old data won't replace the new one
+    let active = true;
+    updateAvailableOffers().then((offers) => {
+      if (active && offers) {
+        setAvailableOffers(offers);
+        setIsLoadingAvailableOffers(false);
+        if (!offers.length) return;
+        const bestOffer: any = offers?.find((offer) => offer.provider == 'Lifi');
+        const selectedOffer = bestOffer?.provider ? bestOffer : offers[0];
+        setSelectedOffer(mapOfferToOption(selectedOffer));
+      }
+    });
+
+    // hook's clean-up function
+    return () => {
+      active = false;
+    };
+  }, [updateAvailableOffers]);
+
+  const availableOffersOptions = useMemo(
+    () => availableOffers?.map(mapOfferToOption),
+    [availableOffers]
+  );
+
+  const receiverAddress = useMemo(() => {
+    if (useCustomAddress) return customReceiverAddress;
+    return selectedReceiveAccountType === AccountTypes.Key
+      ? providerAddress
+      : accountAddress;
+  }, [
+    useCustomAddress,
+    customReceiverAddress,
+    providerAddress,
+    selectedReceiveAccountType,
+    selectedAccountType,
+    accountAddress,
+  ]);
+
+  useEffect(() => {
+    const offer = availableOffers?.find(
+      (availableOffer) => availableOffer.provider === selectedOffer?.value
+    );
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+    if (receiverAddress && !isValidEthereumAddress(receiverAddress)) {
+      setTransactionBlockFieldValidationError(
+        transactionBlockId,
+        'receiverAddress',
+        'Invalid receiver address'
+      );
+      return;
+    }
+    resetTransactionBlockFieldValidationError(
+      transactionBlockId,
+      'receiverAddress'
+    );
+    setTransactionBlockValues(transactionBlockId, {
+      hasEnoughPLR,
+      fromChainId: selectedFromNetwork?.chainId ?? undefined,
+      toAsset: hasEnoughPLR ? plrDaoMemberNFT : plrDaoAsset,
+      fromAsset: selectedFromAsset ?? undefined,
+      amount,
+      offer,
+      accountType: selectedAccountType,
+      receiverAddress: receiverAddress ?? undefined,
+    });
+  }, [
+    selectedFromNetwork,
+    selectedFromAsset,
+    amount,
+    selectedOffer,
+    availableOffers,
+    selectedAccountType,
+    receiverAddress,
+  ]);
```

Comment:
There are a ton of `useEffect` blocks here. Whilst it's not an anti-pattern per se, please add a comment block explaining what each one does as it's clear you've grouped them.